Lib/test/test_exceptions.py

```diff
@@ -1330,6 +1331,38 @@ def test_assert_shadowing(self):
             del AssertionError
             self.fail('Expected exception')
 
+    def test_memory_error_subclasses(self):
+        # MemoryError instances use a freelist of objects that are
+        # linked using the 'dict' attribute when they are
+        # inactive/dead. Subclasses of MemoryError should not
+        # participate in the freelist schema. This test creates a
+        # MemoryError object and keeps it alive (therefore
+        # advancing the freelist) and then it creates and destroys
+        # a subclass object. Finally, it checks that creating a
+        # new MemoryError succeeds, proving that the freelist is
+        # not corrupted. See bpo-41654 for more information.
```

Comment:
"See bpo-41654 for more information.": nitpick: I like to just add "bpo-41654: " at the *beginning* of the comment. But it's up to you.

Reply:
Will change